src/thing/WiFiManager.cpp

```diff
@@ -0,0 +1,69 @@
+#include "thing/WiFiManager.h"
+
+#include <ESP8266WiFi.h>
+#include <DNSServer.h>
+
+
+namespace thing {
+namespace {
+
+const short kDnsPort = 53;
+
+}  // namespace
+
+WiFiManager::WiFiManager() : debug_([](const char*){}) {}
+
+bool WiFiManager::StartAP() {
+  // Use arduino string, casting an int to std::string with arduino C++ support
+  // isn't easy.
+  String ssid("FireThing-");
+  ssid += ESP.getChipId();
+
+  WiFi.mode(WIFI_AP_STA);
+  if (!WiFi.softAP(ssid.c_str())) {
+    return false;
+  }
+
+  debug_((String("WiFi AP : ") + ssid).c_str());
+  debug_((String("Wifi AP IP : ") + WiFi.softAPIP().toString()).c_str());
+
+  dns_.reset(new DNSServer());
```

Comment:
We could leave it running indefinitely without any issues.
 What DNS servers your computer uses is part of the settings that DHCP provides. So if you are a client of the esp's DHCP server you will get this as your dns server (I am not certain how this is specified actually as we never talk directly to the esp's dhcp server but it works). But if you are a client of another DHCP server, like if you are on GoogleGuest, than you will get their dns servers (probably 8.8.8.8, Google's dns server).